### PR TITLE
[NEXMANAGE-826] Update granular log to support arch-independent package

### PR DESCRIPTION
**Pull request by yengliong93**
### PULL DESCRIPTION

Some packages are architecture-independent and are denoted with the suffix "all", for example, linux-firmware:all. In the history.log, it is recorded as linux-firmware:amd64. However, when using the dpkg command, it appears as linux-firmware:all instead of linux-firmware:amd64. This PR updates the granular log to recheck the package status for "all" architecture if the initial attempt fails to retrieve the package status.

More details in NEXMANAGE-826 ticket.

### Impact Analysis

| Info | Please fill out this column |
| ------ | ----------- |
| Root Cause | Architecture-independent package is not supported in granular log handler. |
| Jira ticket | NEXMANAGE-826 |

### CODE MAINTAINABILITY

- [x] Added required new tests relevant to the changes
- [ ] Updated Documentation as relevant to the changes
- [ ] PR change contains code related to security
- [ ] PR introduces changes that break compatibility with other modules/services (If YES, please provide description)
- [ ] Run `go fmt` or `format-python.sh` as applicable
- [x] Update Changelog
- [ ] Integration tests are passing
- [ ] If Cloudadapter changes, check Azure connectivity manually

# _Code must act as a teacher for future developers_

### CODE REVIEW

**Comment by yengliong93** [comment]
@gblewis1  please have a review, thanks.

File: inbm-lib/inbm_lib/package_info.py [hunk 1]
```diff
@@ -68,7 +68,19 @@ def check_package_status(package_name: str) -> str:
...
+        # Some packages are architecture independent.
+        # For example, using dpkg, the package is showing <pacakge>:all instead of <package>:amd64
+        # We will check the status again with independent architecture - "all"
+        try:
+            split_pkg_and_arch = package_name.split(':')
+            arch_independent_pkg = split_pkg_and_arch[0] + ':all'
+            out, err, code = shell.run(
+                "dpkg-query -W -f='${Status}\n' " + arch_independent_pkg)
+            if err:
+                return PACKAGE_FAIL
```
Comment (gblewis1) [inline inbm-lib/inbm_lib/package_info.py]: let's have a new state PACKAGE_UNKNOWN when we can't find the package

Comment (gblewis1) [reply inbm-lib/inbm_lib/package_info.py]: but i'll call this a nice to have, so i will accept this PR as is

**Review by gblewis1 — APPROVED** [review]

File: inbm-lib/inbm_lib/package_info.py (hunk 1 above)
Comment (yengliong93) [reply inbm-lib/inbm_lib/package_info.py]: Updated. Use PACKAGE_UNKNOWN instead of PACKAGE_FAIL when checking package.

**Pull request merged**